core-bundle/src/Controller/AbstractController.php

```diff
@@ -42,4 +45,24 @@ protected function tagResponse(array $tags): void
 
         $this->get('fos_http_cache.http.symfony_response_tagger')->addTags($tags);
     }
+
+    /**
+     * Uses the Symfony router to generate a URL for the given content.
+     * _Content_ in this case should be any supported model/entity of Contao, e.g. a PageModel, NewsModel or similar.
+     */
+    protected function generateContentUrl($content, array $parameters = [], int $referenceType = UrlGeneratorInterface::ABSOLUTE_PATH): string
+    {
+        $parameters[PageRoute::CONTENT_PARAMETER] = $content;
+
+        return $this->generateUrl(PageRoute::ROUTE_NAME, $parameters, $referenceType);
+    }
+
+    /**
+     * Uses the Symfony router to generate redirect response to the URL of the given content.
+     * _Content_ in this case should be any supported model/entity of Contao, e.g. a PageModel, NewsModel or similar.
+     */
+    protected function redirectToContent($content, array $parameters = [], int $status = 302): RedirectResponse
```

Comment:
Can we add a `@param` annotation for the `$content` argument? There is no interface for Doctrine entities, though, that's probably why you have not added an annotation yet.

Reply:
The annotation was removed by the CS-Fixer for me, probably because it did not provide any helpful information ( (`@param mixed $content`).

Reply:
Yes, if it was `mixed` then it was correct to remove it. I was thinking about something like `@param Model|DoctrineEntity` – except there is no parent class for Doctrine entities. 🤷‍♂️

Reply:
Well, every class could be an entity. What *property* of an entity is required here?